src/github.com/matrix-org/dendrite/roomserver/query/query.go

```diff
@@ -487,6 +487,80 @@ BFSLoop:
 	return err
 }
 
+// QueryPreviousEvents implements api.RoomServerQueryAPI
+func (r *RoomserverQueryAPI) QueryPreviousEvents(
```

Comment:
`QueryPreviousEvents` feels a little overly generic. Can we call it like `QueryForBackfill` or something? I'm expecting eventually we'll want to optimise the query to be more efficient (BFS is fine for now, but isn't the best)